3_javascript/javascript1/week3/Homework/script.js

```diff
@@ -0,0 +1,219 @@
+// warm up work item array removal //
+
+const names = [
+  "Peter",
+  "Ahmad",
+  "Yana",
+  "kristina",
+  "Rasmus",
+  "Samuel",
+  "katrine",
+  "Tala",
+];
+const nameToRemove = "Ahmad";
+
+// Write some code here
+
+console.log(names.at(1));
+
+// trying other parameters //
+
+console.log(names.indexOf("Rasmus"));
+
+console.log(names[names.length - 1]);
+
+// Code done
+
+console.log(names); // ['Peter', 'Yana', 'kristina', 'Rasmus', 'Samuel', 'katrine', 'Tala']
+
+//  When will we be there //
+
+const travelInformation = {
+  speed: 50,
+  destinationDistance: 432,
+};
+
+const averageSpeedToArrive = function (num) {
+  let hours = Math.floor(num / 60);
+  if (hours.toString().length === 1) {
+    hours = "0" + hours;
+  }
+
+  let minutes = num % 60;
+  if (minutes.toString().length === 1) {
+    minutes = "0" + minutes;
+  }
+
+  return `${hours}:${minutes}`;
+};
+
+const travelTime = averageSpeedToArrive(travelInformation);
+console.log(averageSpeedToArrive(432)); // 8 hours and 38 minutes
+
+// Series duration of my life //
+
+const seriesDurations = [
+  {
+    title: "Friends",
+    days: 3,
+    hours: 23,
+    minutes: 0,
+  },
+  {
+    title: "Hannah Montana",
+    days: 5,
+    hours: 12,
+    minutes: 57,
+  },
+  {
+    title: "Harry Potter",
+    days: 5,
+    hours: 12,
+    minutes: 57,
+  },
+];
+
+const seriesDurationsOfMyLife = function (seriesDuration) {
+  // write code here
+
+  let output = "";
+  for (seriesDuration of seriesDuration) {
+    let totalMinutesOfSeries =
+      seriesDuration.days * 24 * 60 +
+      seriesDuration.hours * 60 +
+      seriesDuration.minutes;
+    const myAverageLifeInMinuets = 80 * 365 * 24 * 60;
+    let percentage = (totalMinutesOfSeries / myAverageLifeInMinuets) * 100;
+    output += `${seriesDuration.title} took ${percentage.toFixed(
+      3
+    )}% of my life.\n`;
+  }
+  return output;
+};
+
+console.log(seriesDurationsOfMyLife(seriesDurations)); // logs out the text found above
+
+// NOnoN0nOYes (Note taking app) //
+
+const notes = [];
+
+function saveNote(content, id) {
+  // write some code here
+  notes.push({
+    content: content,
+    id: id,
+  });
+}
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+
+console.log(notes); // [{content: 'Pick up groceries', id: 1}, {content: 'Do laundry', id: 2}]
+
+// Get a note //
+
+function getNote(id) {
+  // your code here
+
+  for (let note of notes) {
+    if (id === note.id) return note.content;
```

Comment:
good job , 
what if the id is not exist in the our notes . we should show some message to user